Wangscape/tilegen/partition/TilePartitionerGradient.cpp

```diff
@@ -25,18 +25,18 @@ void TilePartitionerGradient::makePartition(TilePartition & regions,
                    mOptions.tileFormat.resolution);
         masks.push_back(img);
     }
-    std::vector<float> weights(corners.size(), 0.f);
+    std::vector<double> weights(corners.size(), 0.);
     const int resolution_sub_1 = mOptions.tileFormat.resolution - 1;
     const int quarter_res = mOptions.tileFormat.resolution / 4;
     alpha::AlphaCalculatorLinear ac;
     for (size_t x = 0; x < mOptions.tileFormat.resolution; x++)
     {
         for (size_t y = 0; y < mOptions.tileFormat.resolution; y++)
         {
-            weights[0] = (float)gradientWeight(x, y, 0, 0, quarter_res);
-            weights[1] = (float)gradientWeight(x, y, 0, resolution_sub_1, quarter_res);
-            weights[2] = (float)gradientWeight(x, y, resolution_sub_1, 0, quarter_res);
-            weights[3] = (float)gradientWeight(x, y, resolution_sub_1, resolution_sub_1, quarter_res);
+            weights[0] = (double)gradientWeight(x, y, 0, 0, quarter_res);
+            weights[1] = (double)gradientWeight(x, y, 0, resolution_sub_1, quarter_res);
+            weights[2] = (double)gradientWeight(x, y, resolution_sub_1, 0, quarter_res);
+            weights[3] = (double)gradientWeight(x, y, resolution_sub_1, resolution_sub_1, quarter_res);
```

Comment:
I'm not sure if it must be casted, but if it does, use `static_cast<double>` here, please. One advantage is that it's easy to grep for, and the other is that it is safer (e.g. another version of "raw casting", `double(...)` can be treated by a compiler as a function declaration).